Submitted By: Mario Fetka (geos_one) (mario dot fetka at gmail dot com)
Date: 2010-02-14
Initial Package Version: 4.0.0
Origin: gentoo patch dhcp-4.0-no_isc_blurb.patch
Upstream Status: 
Description: 
 
diff -Naur dhcp-4.0.2.orig/client/dhclient.c dhcp-4.0.2/client/dhclient.c
--- dhcp-4.0.2.orig/client/dhclient.c	2010-02-14 07:26:37.018800468 +0000
+++ dhcp-4.0.2/client/dhclient.c	2010-02-14 07:26:44.163801138 +0000
@@ -80,6 +80,8 @@
 
 static isc_result_t write_duid(struct data_string *duid);
 
+extern int log_isc_blurb;
+
 int 
 main(int argc, char **argv) {
 	int fd;
@@ -325,6 +327,7 @@
 		log_info ("%s", "");
 	} else {
 		log_perror = 0;
+		log_isc_blurb = 0;
 		quiet_interface_discovery = 1;
 	}
 
diff -Naur dhcp-4.0.2.orig/omapip/errwarn.c dhcp-4.0.2/omapip/errwarn.c
--- dhcp-4.0.2.orig/omapip/errwarn.c	2010-02-14 07:26:37.006801498 +0000
+++ dhcp-4.0.2/omapip/errwarn.c	2010-02-14 07:26:44.164800970 +0000
@@ -43,6 +43,8 @@
 int log_perror = 1;
 #endif
 int log_priority;
+int log_isc_blurb=1;
+
 void (*log_cleanup) (void);
 
 #define CVT_BUF_MAX 1023
@@ -74,7 +76,9 @@
 	  IGNORE_RET (write (STDERR_FILENO, "\n", 1));
   }
 
-#if !defined (NOMINUM)
+#if !defined(NOMINUM)
+  if ( log_isc_blurb )
+  {
   log_error ("%s", "");
   log_error ("If you did not get this software from ftp.isc.org, please");
   log_error ("get the latest from ftp.isc.org and install that before");
@@ -92,7 +96,12 @@
   log_error ("the README file.");
   log_error ("%s", "");
   log_error ("exiting.");
+  }else
+  {
+      log_error ("exiting.");
+  }
 #endif
+
   if (log_cleanup)
 	  (*log_cleanup) ();
   exit (1);
diff -Naur dhcp-4.0.2.orig/relay/dhcrelay.c dhcp-4.0.2/relay/dhcrelay.c
--- dhcp-4.0.2.orig/relay/dhcrelay.c	2010-02-14 07:26:37.014800648 +0000
+++ dhcp-4.0.2/relay/dhcrelay.c	2010-02-14 07:28:00.771800735 +0000
@@ -103,7 +103,8 @@
 "Internet Systems Consortium DHCP Relay Agent";
 static const char url[] =
 "For info, please visit https://www.isc.org/software/dhcp/";
-  
+extern int log_isc_blurb;
+
 int 
 main(int argc, char **argv) {
 	int fd;
@@ -169,6 +170,7 @@
 		} else if (!strcmp (argv [i], "-q")) {
 			quiet = 1;
 			quiet_interface_discovery = 1;
+			log_isc_blurb = 0;
 		} else if (!strcmp (argv [i], "-a")) {
 			add_agent_options = 1;
 		} else if (!strcmp (argv [i], "-c")) {
diff -Naur dhcp-4.0.2.orig/server/dhcpd.c dhcp-4.0.2/server/dhcpd.c
--- dhcp-4.0.2.orig/server/dhcpd.c	2010-02-14 07:26:37.047800993 +0000
+++ dhcp-4.0.2/server/dhcpd.c	2010-02-14 07:26:44.165800803 +0000
@@ -62,6 +62,9 @@
 struct iaddr server_identifier;
 int server_identifier_matched;
 
+
+extern int log_isc_blurb;
+
 #if defined (NSUPDATE)
 
 /* This stuff is always executed to figure the default values for certain
@@ -354,6 +357,7 @@
 			lftest = 1;
 			log_perror = -1;
 		} else if (!strcmp (argv [i], "-q")) {
+			log_isc_blurb = 0;
 			quiet = 1;
 			quiet_interface_discovery = 1;
 #ifdef DHCPv6
